pkg/services/rpcsrv/subscription_test.go

```diff
@@ -376,7 +395,16 @@ func TestFilteredSubscriptions(t *testing.T) {
 					}
 					continue
 				}
-				this.check(t, resp)
+				if this.shouldCheck {
+					checked = true
+					this.check(t, resp)
+				} else {
+					t.Fatal("unexpected notification: " + resp.EventID().String())
+				}
+			}
+
+			if this.shouldCheck && !checked {
+				t.Fatalf("expected check is not performed")
```

Comment:
`t.Fatalf` is used when you need a formatted error message. You have only a single string here.